(hdt-lstm) C:\Users\USER-PC\Desktop\HumanDigitalTwin_LSTM\Training Iteration #10 - ANN time series>python Train10.py
2025-07-09 01:38:12.317934: I tensorflow/core/util/port.cc:153] oneDNN custom operations are on. You may see slightly different numerical results due to floating-point round-off errors from different computation orders. To turn them off, set the environment variable `TF_ENABLE_ONEDNN_OPTS=0`.
2025-07-09 01:38:13.814545: I tensorflow/core/util/port.cc:153] oneDNN custom operations are on. You may see slightly different numerical results due to floating-point round-off errors from different computation orders. To turn them off, set the environment variable `TF_ENABLE_ONEDNN_OPTS=0`.
Detected 71 feature columns.
2025-07-09 01:38:17.748948: I tensorflow/core/platform/cpu_feature_guard.cc:210] This TensorFlow binary is optimized to use available CPU instructions in performance-critical operations.
To enable the following instructions: SSE3 SSE4.1 SSE4.2 AVX AVX2 FMA, in other operations, rebuild TensorFlow with the appropriate compiler flags.
C:\Users\USER-PC\miniconda3\envs\hdt-lstm\lib\site-packages\keras\src\layers\rnn\rnn.py:199: UserWarning: Do not pass an `input_shape`/`input_dim` argument to a layer. When using Sequential models, prefer using an `Input(shape)` object as the first layer in the model instead.
  super().__init__(**kwargs)

=== Starting training... ===
Epoch 1/50
100/100 - 8s - 78ms/step - accuracy: 0.4642 - loss: 1.2114 - learning_rate: 1.0000e-03
Epoch 2/50
100/100 - 6s - 61ms/step - accuracy: 0.4025 - loss: 1.2690 - learning_rate: 1.0000e-03
Epoch 3/50
100/100 - 6s - 61ms/step - accuracy: 0.4068 - loss: 1.2422 - learning_rate: 1.0000e-03
Epoch 4/50
100/100 - 6s - 61ms/step - accuracy: 0.4461 - loss: 1.2174 - learning_rate: 1.0000e-03
Epoch 5/50
100/100 - 6s - 61ms/step - accuracy: 0.4664 - loss: 1.1933 - learning_rate: 5.0000e-04
Epoch 6/50
100/100 - 6s - 61ms/step - accuracy: 0.4703 - loss: 1.2132 - learning_rate: 5.0000e-04
Epoch 7/50
100/100 - 6s - 61ms/step - accuracy: 0.4674 - loss: 1.2061 - learning_rate: 5.0000e-04
Epoch 8/50
100/100 - 6s - 61ms/step - accuracy: 0.4419 - loss: 1.1825 - learning_rate: 5.0000e-04
Epoch 9/50
100/100 - 6s - 62ms/step - accuracy: 0.4786 - loss: 1.1561 - learning_rate: 5.0000e-04
Epoch 10/50
100/100 - 6s - 61ms/step - accuracy: 0.4658 - loss: 1.1655 - learning_rate: 5.0000e-04
Epoch 11/50
100/100 - 6s - 61ms/step - accuracy: 0.4648 - loss: 1.2014 - learning_rate: 5.0000e-04
Epoch 12/50
100/100 - 6s - 61ms/step - accuracy: 0.4468 - loss: 1.2394 - learning_rate: 5.0000e-04
Epoch 13/50
100/100 - 6s - 61ms/step - accuracy: 0.4510 - loss: 1.1908 - learning_rate: 2.5000e-04
Epoch 14/50
100/100 - 6s - 61ms/step - accuracy: 0.4725 - loss: 1.1651 - learning_rate: 2.5000e-04
Epoch 15/50
100/100 - 6s - 61ms/step - accuracy: 0.4456 - loss: 1.2040 - learning_rate: 2.5000e-04
Epoch 16/50
100/100 - 6s - 61ms/step - accuracy: 0.4671 - loss: 1.1676 - learning_rate: 1.2500e-04
Epoch 17/50
100/100 - 6s - 61ms/step - accuracy: 0.4464 - loss: 1.2429 - learning_rate: 1.2500e-04
Epoch 18/50
2025-07-09 01:40:07.337105: I tensorflow/core/framework/local_rendezvous.cc:407] Local rendezvous is aborting with status: OUT_OF_RANGE: End of sequence
         [[{{node IteratorGetNext}}]]
C:\Users\USER-PC\miniconda3\envs\hdt-lstm\lib\site-packages\keras\src\trainers\epoch_iterator.py:160: UserWarning: Your input ran out of data; interrupting training. Make sure that your dataset or generator can generate at least `steps_per_epoch * epochs` batches. You may need to use the `.repeat()` function when building your dataset.
  self._interrupted_warning()
100/100 - 4s - 37ms/step - accuracy: 0.4243 - loss: 1.2849 - learning_rate: 1.2500e-04
Epoch 19/50
2025-07-09 01:40:07.455335: I tensorflow/core/framework/local_rendezvous.cc:407] Local rendezvous is aborting with status: OUT_OF_RANGE: End of sequence
         [[{{node IteratorGetNext}}]]
100/100 - 0s - 1ms/step - accuracy: 0.8613 - loss: 0.8798 - learning_rate: 6.2500e-05
Epoch 20/50
100/100 - 0s - 1ms/step - accuracy: 0.8617 - loss: 0.8777 - learning_rate: 6.2500e-05
Epoch 21/50
2025-07-09 01:40:07.687319: I tensorflow/core/framework/local_rendezvous.cc:407] Local rendezvous is aborting with status: OUT_OF_RANGE: End of sequence
         [[{{node IteratorGetNext}}]]
100/100 - 0s - 1ms/step - accuracy: 0.8618 - loss: 0.8734 - learning_rate: 6.2500e-05
Epoch 22/50
100/100 - 0s - 1ms/step - accuracy: 0.8619 - loss: 0.8669 - learning_rate: 6.2500e-05
Epoch 23/50
100/100 - 0s - 1ms/step - accuracy: 0.8625 - loss: 0.8589 - learning_rate: 6.2500e-05
Epoch 24/50
100/100 - 0s - 1ms/step - accuracy: 0.8621 - loss: 0.8507 - learning_rate: 6.2500e-05
Epoch 25/50
2025-07-09 01:40:08.147585: I tensorflow/core/framework/local_rendezvous.cc:407] Local rendezvous is aborting with status: OUT_OF_RANGE: End of sequence
         [[{{node IteratorGetNext}}]]
100/100 - 0s - 1ms/step - accuracy: 0.8630 - loss: 0.8410 - learning_rate: 6.2500e-05
Epoch 26/50
100/100 - 0s - 1ms/step - accuracy: 0.8634 - loss: 0.8316 - learning_rate: 6.2500e-05
Epoch 27/50
100/100 - 0s - 1ms/step - accuracy: 0.8635 - loss: 0.8211 - learning_rate: 6.2500e-05
Epoch 28/50
100/100 - 0s - 1ms/step - accuracy: 0.8643 - loss: 0.8107 - learning_rate: 6.2500e-05
Epoch 29/50
100/100 - 0s - 1ms/step - accuracy: 0.8639 - loss: 0.8021 - learning_rate: 6.2500e-05
Epoch 30/50
100/100 - 0s - 1ms/step - accuracy: 0.8647 - loss: 0.7901 - learning_rate: 6.2500e-05
Epoch 31/50
100/100 - 0s - 1ms/step - accuracy: 0.8648 - loss: 0.7800 - learning_rate: 6.2500e-05
Epoch 32/50
100/100 - 0s - 1ms/step - accuracy: 0.8647 - loss: 0.7704 - learning_rate: 6.2500e-05
Epoch 33/50
2025-07-09 01:40:09.065256: I tensorflow/core/framework/local_rendezvous.cc:407] Local rendezvous is aborting with status: OUT_OF_RANGE: End of sequence
         [[{{node IteratorGetNext}}]]
100/100 - 0s - 1ms/step - accuracy: 0.8648 - loss: 0.7596 - learning_rate: 6.2500e-05
Epoch 34/50
100/100 - 0s - 1ms/step - accuracy: 0.8647 - loss: 0.7508 - learning_rate: 6.2500e-05
Epoch 35/50
100/100 - 0s - 1ms/step - accuracy: 0.8649 - loss: 0.7402 - learning_rate: 6.2500e-05
Epoch 36/50
100/100 - 0s - 1ms/step - accuracy: 0.8650 - loss: 0.7317 - learning_rate: 6.2500e-05
Epoch 37/50
100/100 - 0s - 1ms/step - accuracy: 0.8650 - loss: 0.7222 - learning_rate: 6.2500e-05
Epoch 38/50
100/100 - 0s - 1ms/step - accuracy: 0.8650 - loss: 0.7132 - learning_rate: 6.2500e-05
Epoch 39/50
100/100 - 0s - 1ms/step - accuracy: 0.8650 - loss: 0.7029 - learning_rate: 6.2500e-05
Epoch 40/50
100/100 - 0s - 1ms/step - accuracy: 0.8650 - loss: 0.6952 - learning_rate: 6.2500e-05
Epoch 41/50
100/100 - 0s - 1ms/step - accuracy: 0.8650 - loss: 0.6848 - learning_rate: 6.2500e-05
Epoch 42/50
100/100 - 0s - 1ms/step - accuracy: 0.8650 - loss: 0.6796 - learning_rate: 6.2500e-05
Epoch 43/50
100/100 - 0s - 1ms/step - accuracy: 0.8650 - loss: 0.6697 - learning_rate: 6.2500e-05
Epoch 44/50
100/100 - 0s - 1ms/step - accuracy: 0.8650 - loss: 0.6627 - learning_rate: 6.2500e-05
Epoch 45/50
100/100 - 0s - 1ms/step - accuracy: 0.8650 - loss: 0.6547 - learning_rate: 6.2500e-05
Epoch 46/50
100/100 - 0s - 1ms/step - accuracy: 0.8650 - loss: 0.6485 - learning_rate: 6.2500e-05
Epoch 47/50
100/100 - 0s - 1ms/step - accuracy: 0.8650 - loss: 0.6408 - learning_rate: 6.2500e-05
Epoch 48/50
100/100 - 0s - 1ms/step - accuracy: 0.8650 - loss: 0.6353 - learning_rate: 6.2500e-05
Epoch 49/50
2025-07-09 01:40:10.905970: I tensorflow/core/framework/local_rendezvous.cc:407] Local rendezvous is aborting with status: OUT_OF_RANGE: End of sequence
         [[{{node IteratorGetNext}}]]
100/100 - 0s - 1ms/step - accuracy: 0.8650 - loss: 0.6270 - learning_rate: 6.2500e-05
Epoch 50/50
100/100 - 0s - 1ms/step - accuracy: 0.8650 - loss: 0.6217 - learning_rate: 6.2500e-05

=== Starting evaluation over the entire dataset… ===
Evaluating: 100%|██████████████████████████████████████████████████████████████████| 1761/1761 [09:04<00:00,  3.24it/s]
C:\Users\USER-PC\miniconda3\envs\hdt-lstm\lib\site-packages\sklearn\metrics\_classification.py:1706: UndefinedMetricWarning: Precision is ill-defined and being set to 0.0 in labels with no predicted samples. Use `zero_division` parameter to control this behavior.
  _warn_prf(average, modifier, f"{metric.capitalize()} is", result.shape[0])
C:\Users\USER-PC\miniconda3\envs\hdt-lstm\lib\site-packages\sklearn\metrics\_classification.py:1706: UndefinedMetricWarning: Precision is ill-defined and being set to 0.0 in labels with no predicted samples. Use `zero_division` parameter to control this behavior.
  _warn_prf(average, modifier, f"{metric.capitalize()} is", result.shape[0])
C:\Users\USER-PC\miniconda3\envs\hdt-lstm\lib\site-packages\sklearn\metrics\_classification.py:1706: UndefinedMetricWarning: Precision is ill-defined and being set to 0.0 in labels with no predicted samples. Use `zero_division` parameter to control this behavior.
  _warn_prf(average, modifier, f"{metric.capitalize()} is", result.shape[0])

======== Model Evaluation Report ========
Accuracy       : 0.4170
ROC AUC (ovr)  : 0.6664

Classification Report:
               precision    recall  f1-score   support

   Withdrawn       0.62      0.13      0.21   2742120
        Fail       0.38      0.08      0.14   1904040
        Pass       0.40      0.95      0.57   3337470
 Distinction       0.00      0.00      0.00    816480

    accuracy                           0.42   8800110
   macro avg       0.35      0.29      0.23   8800110
weighted avg       0.43      0.42      0.31   8800110

Confusion Matrix:
 [[ 352452  170041 2219627       0]
 [ 104526  160692 1638822       0]
 [  93444   87500 3156526       0]
 [  14484    6882  795114       0]]
Saved artifact at 'C:\Users\USER-PC\Desktop\HumanDigitalTwin_LSTM\Training Iteration #10 - ANN time series\Train10Model'. The following endpoints are available:

* Endpoint 'serve'
  args_0 (POSITIONAL_ONLY): TensorSpec(shape=(None, 1, 71), dtype=tf.float32, name='keras_tensor')
Output Type:
  TensorSpec(shape=(None, 4), dtype=tf.float32, name=None)
Captures:
  2111304353744: TensorSpec(shape=(), dtype=tf.resource, name=None)
  2111304357616: TensorSpec(shape=(), dtype=tf.resource, name=None)
  2111304366064: TensorSpec(shape=(), dtype=tf.resource, name=None)
  2111304364128: TensorSpec(shape=(), dtype=tf.resource, name=None)
  2111304353568: TensorSpec(shape=(), dtype=tf.resource, name=None)
  2111304358144: TensorSpec(shape=(), dtype=tf.resource, name=None)
  2111304280272: TensorSpec(shape=(), dtype=tf.resource, name=None)

Model exported to: C:\Users\USER-PC\Desktop\HumanDigitalTwin_LSTM\Training Iteration #10 - ANN time series\Train10Model

(hdt-lstm) C:\Users\USER-PC\Desktop\HumanDigitalTwin_LSTM\Training Iteration #10 - ANN time series>